tools/detect_fsgsbase/detect_fsgsbase.c

```diff
@@ -0,0 +1,48 @@
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <signal.h>
+#include <assert.h>
+#include <string.h>
+#include <ucontext.h>
+#include <setjmp.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#define RC 0xffff
+jmp_buf env_buf;
+
+static void handle_sigill(int num) {
+    printf("\tSIGILL Caught !\n");
+    assert(num == SIGILL);
+    printf("\tFSGSBASE instructions are not supported.\n");
+
+    // since we don't know how far we should jump, just use longjmp to exit
+    longjmp(env_buf, RC);
+}
+
+int main (void) {
```

Comment:
Since the code is very simple, should we add it in the occlum_pal.so just before launch the occlum instance?

Reply:
Agree with @guzongmin . Let's add it to PAL.